pdf_compression/models/report.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api
+from openerp.exceptions import AccessError
+from openerp.osv import osv
+from openerp.tools.translate import _
+from openerp.addons.web.http import request
+from openerp.addons.report.models.report import _get_wkhtmltopdf_bin
+
+import base64
+import logging
+import tempfile
+import os
+import subprocess
+from contextlib import closing
+
+_logger = logging.getLogger(__name__)
+
+
+class Report(osv.Model):
+    _inherit = "report"
+
+    @api.v8
+    def get_pdf(self, records, report_name, html=None, data=None):
+        """This method generates and returns pdf version of a report.
+        """
+        ctx = dict(self._context)
+        report = self._get_report_from_name(report_name)
+        ctx.update({'pdf_option': report.pdf_option})
+        return super(Report, self.with_context(ctx)).get_pdf(
+            records, report_name, html=html, data=data)
+
+    @api.v7
+    def get_pdf(self, cr, uid, ids, report_name, html=None, data=None,
+                context=None):
+        """override to add pdf_option to the context"""
+        ctx = dict(context)
+        report = self._get_report_from_name(cr, uid, report_name)
+        ctx.update({'pdf_option': report.pdf_option})
+        return super(Report, self).get_pdf(
+            cr, uid, ids, report_name, html=html, data=data, context=context)
+
+    def _run_wkhtmltopdf(self, cr, uid, headers, footers, bodies, landscape,
```

Comment:
This looks to be identical to the core method - https://github.com/odoo/odoo/blob/8.0/addons/report/models/report.py#L370